code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -36,6 +36,58 @@
 	projectile_type = /obj/item/projectile/beam
 	charge_cost = 50
 
+/obj/item/weapon/gun/energy/laser/failure_check(var/mob/living/carbon/human/M)
```

Comment:
You're constantly copy pasting a lot of stuff like the message of it popping, in chamber being reset, and the firing delay. Try to generalize it some more.